src/software/ai/hl/stp/tactic/receiver/receiver_fsm.h

```diff
@@ -0,0 +1,296 @@
+#pragma once
+
+#include "shared/constants.h"
+#include "software/ai/evaluation/calc_best_shot.h"
+#include "software/ai/hl/stp/tactic/dribble/dribble_fsm.h"
+#include "software/ai/hl/stp/tactic/kick/kick_fsm.h"
+#include "software/ai/hl/stp/tactic/move/move_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/intent.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/ai/passing/pass.h"
+#include "software/geom/algorithms/closest_point.h"
+
+struct ReceiverFSM
+{
+    class OneTouchShotState;
+    class ReceiveAndDribbleState;
+    class UndecidedState;
+
+    struct ControlParams
+    {
+        // The pass to receive
+        std::optional<Pass> pass = std::nullopt;
+    };
+
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    // The minimum proportion of open net we're shooting on vs the entire size of the net
+    // that we require before attempting a shot
+    static constexpr double MIN_SHOT_NET_PERCENT_OPEN    = 0.3;
+    static constexpr double MAX_SPEED_FOR_ONE_TOUCH_SHOT = 6.5;
```

Comment:
can't this just be max kick speed from the rules?

Reply:
it is rn, i think maybe it needs to be slower if our robots can't damp the ball enough at that speed IRL

but we will cross that bridge when we get there, updated to just use max kick speed

Reply:
hm, this should be a robot constant